#365711 causes EGL to break in sway when using nvidia

## Describe the bug

#365711 causes EGL to break in sway when using nvidia

## Steps To Reproduce

Steps to reproduce the behavior:

`services.xserver.videoDrivers = [ "nvidia" ];`
`hardware.nvidia.open = false;`
`programs.sway.enable = true;`

## Expected behavior

EGL works, and applications that use it benefit from hardware acceleration.

## Screenshots

I found the cause of this issue by bisecting nixos, here's a screenshot of the middle of bisecting, with examples of errors that occur when EGL is broken:
- sway: the errors that start with `[ERROR] [wlr] [EGL]` `[ERROR] [wlr] [render/egl.c:851]`
- mpv: the first warning that starts with `[vo/gpu-next/opengl]` (the rest are various fallbacks failing until finding `waylandvk` which works)

![20250107_04h45m26s_grim](https://github.com/user-attachments/assets/38330218-670e-44fb-96cd-54578ebff54a)
Other effects include `eglinfo` reporting a software renderer on the Wayland platform, and screen sharing not working in firefox.

## Additional context

The problem is easily solved by setting:
`hardware.nvidia.package = config.boot.kernelPackages.nvidiaPackages.latest;`
I don't disagree with the change in #365711, but I'm a bit upset that it took bisecting nixos to discover this change and find this easy fix.

## Metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.69, NixOS, 25.05 (Warbler), 25.05beta733374.8f3e1f807051`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.11`
 - channels(root): `"nixos"`
 - channels(erry): `"home-manager"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

## Notify maintainers

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

@NklsCh 

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/nixos/doc/manual/release-notes/rl-2505.section.md
+++ b/nixos/doc/manual/release-notes/rl-2505.section.md
@@ -132,6 +132,8 @@
   +services.asusd.asusdConfig.text = '''file contents'''
   ```
 
+- `linuxPackages.nvidiaPackages.stable` now defaults to the `production` variant instead of `latest`.
+
 - `timescaledb` requires manual upgrade steps.
     After you run ALTER EXTENSION, you must run [this SQL script](https://github.com/timescale/timescaledb-extras/blob/master/utils/2.15.X-fix_hypertable_foreign_keys.sql). For more details, see the following pull requests [#6797](https://github.com/timescale/timescaledb/pull/6797).
     PostgreSQL 13 is no longer supported in TimescaleDB v2.16.

--- a/nixos/modules/programs/wayland/sway.nix
+++ b/nixos/modules/programs/wayland/sway.nix
@@ -113,6 +113,19 @@ in
         }
       ];
 
+      warnings =
+          lib.mkIf
+            (
+              (lib.elem "nvidia" config.services.xserver.videoDrivers)
+              && !config.hardware.nvidia.open
+              && (lib.versionOlder "551" (
+                lib.versions.major (lib.getVersion config.hardware.nvidia.package)
+              ))
+            )
+            [
+              "Using Sway with Nvidia driver version <= 550 may result in a broken system. Configure hardware.nvidia.package to use a newer version."
+            ];
+
       environment = {
         systemPackages = lib.optional (cfg.package != null) cfg.package ++ cfg.extraPackages;
 

